core/templates/dev/head/pages/exploration_player/CurrentInteractionService.js

```diff
@@ -0,0 +1,100 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Facilitates communication between the current interaction
+ * and the progress nav. The former holds data about the learner's answer,
+ * while the latter contains the actual "Submit" button which triggers the
+ * answer submission process.
+ */
+
+oppia.factory('CurrentInteractionService', [
```

Comment:
This service should have Karma tests (to illustrate its use and prevent future regressions).

Reply:
Done.